nav2_util/src/robot_utils.cpp

```diff
@@ -143,6 +143,66 @@ bool getTransform(
   return true;
 }
 
+bool getTransform(
+  const std::string & source_frame_id,
+  const std::string & target_frame_id,
+  const tf2::Duration & transform_tolerance,
+  const std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+  geometry_msgs::msg::TransformStamped & transform_stamped)
+{
+  geometry_msgs::msg::TransformStamped transform;
+
+  if (source_frame_id == target_frame_id) {
+    // We are already in required frame
+    return true;
+  }
+
+  try {
+    // Obtaining the transform to get data from source to target frame
+    transform_stamped = tf_buffer->lookupTransform(
+      target_frame_id, source_frame_id,
+      tf2::TimePointZero, transform_tolerance);
+  } catch (tf2::TransformException & e) {
+    RCLCPP_ERROR(
+      rclcpp::get_logger("getTransform"),
+      "Failed to get \"%s\"->\"%s\" frame transform: %s",
+      source_frame_id.c_str(), target_frame_id.c_str(), e.what());
+    return false;
+  }
+
+  return true;
+}
+
+bool getTransform(
+  const std::string & source_frame_id,
+  const rclcpp::Time & source_time,
+  const std::string & target_frame_id,
+  const rclcpp::Time & target_time,
+  const std::string & fixed_frame_id,
+  const tf2::Duration & transform_tolerance,
+  const std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+  geometry_msgs::msg::TransformStamped & transform_stamped)
```

Comment:
I'd also like to get rid of duplicate code, as @SteveMacenski mentioned by utilizing `tf2::Transform`. Another alternative could be making this routine as inline and calling it from inside non-stamped `getTransform` version

Reply:
I think I found a satisfying solution, I changed the output of `getTransform` from `tf2::Transform` to `tf2::Stamped<tf2::Transform>` so that we can keep the stamp information. The other sources are not affected since `    tf2::Vector3 p_v3_b = tf_transform * p_v3_s;` should work with both stamped and unstamped Transform